src/ssl.c

```diff
@@ -3889,6 +3889,53 @@ WOLFSSL_STACK* wolfSSL_CertManagerGetCerts(WOLFSSL_CERT_MANAGER* cm)
 
     return NULL;
 }
+
+WOLF_STACK_OF(WOLFSSL_X509)* wolfSSL_X509_STORE_get1_certs(
+    WOLFSSL_X509_STORE_CTX* ctx, WOLFSSL_X509_NAME* name)
+{
+    WOLFSSL_ENTER("wolfSSL_X509_STORE_get1_certs");
+    WOLFSSL_X509_STORE* store;
+    WOLFSSL_STACK* sk;
+    WOLFSSL_STACK* certToFilter;
+    WOLFSSL_X509_NAME* certToFilterName;
+    WOLF_STACK_OF(WOLFSSL_X509)* filteredCerts = NULL;
+    WOLFSSL_X509* filteredCert = NULL;
+
+    if (name == NULL)
+        return NULL;
+
+    store = wolfSSL_X509_STORE_CTX_get0_store(ctx);
+    if (store == NULL)
+        return NULL;
+
+    filteredCerts = wolfSSL_sk_X509_new();
+    if (filteredCerts == NULL)
+        return NULL;
+
+    sk = wolfSSL_CertManagerGetCerts(store->cm);
```

Comment:
Move above allocating cert so that you don't have to free it on error.

Reply:
Re-wrote this function to have one return statement. Simplifies some of the freeing logic.